# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1688284859 -7200
# Parent  ea887f002aae03694e27be639b926725decbb18d
Feature: pref mail.biff.show_for_server to disable biff display in tree for servers.

See https://bugzilla.mozilla.org/show_bug.cgi?id=1545955#c5 for some details.

diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -2278,16 +2278,22 @@ var folderPane = {
 
   /**
    * Called when a folder's new messages state changes.
    *
    * @param {nsIMsgFolder} folder
    * @param {boolean} hasNewMessages
    */
   changeNewMessages(folder, hasNewMessages) {
+    if (
+      folder.isServer &&
+      !Services.prefs.getBoolPref("mail.biff.show_for_server", true)
+    ) {
+      return;
+    }
     this._changeRows(folder, row =>
       row.classList.toggle("new-messages", hasNewMessages)
     );
   },
 
   /**
    * Called when a folder's unread count changes, to update the UI.
    *
diff --git a/mailnews/base/src/FolderUtils.jsm b/mailnews/base/src/FolderUtils.jsm
--- a/mailnews/base/src/FolderUtils.jsm
+++ b/mailnews/base/src/FolderUtils.jsm
@@ -78,26 +78,29 @@ function getFolderProperties(aFolder, aO
 
   properties.push("folderNameCol");
 
   properties.push("serverType-" + aFolder.server.type);
 
   // set the SpecialFolder attribute
   properties.push("specialFolder-" + getSpecialFolderString(aFolder));
 
-  // Now set the biffState
-  switch (aFolder.biffState) {
-    case nsIMsgFolder.nsMsgBiffState_NewMail:
-      properties.push("biffState-NewMail");
-      break;
-    case nsIMsgFolder.nsMsgBiffState_NoMail:
-      properties.push("biffState-NoMail");
-      break;
-    default:
-      properties.push("biffState-UnknownMail");
+  if (aFolder.isServer &&
+      Services.prefs.getBoolPref("mail.biff.show_for_server", true)) {
+    // Now set the biffState
+    switch (aFolder.biffState) {
+      case nsIMsgFolder.nsMsgBiffState_NewMail:
+        properties.push("biffState-NewMail");
+        break;
+      case nsIMsgFolder.nsMsgBiffState_NoMail:
+        properties.push("biffState-NoMail");
+        break;
+      default:
+        properties.push("biffState-UnknownMail");
+    }
   }
 
   properties.push("isSecure-" + aFolder.server.isSecure);
 
   // A folder has new messages, or a closed folder or any subfolder has new messages.
   if (
     aFolder.hasNewMessages ||
     (!aOpen && aFolder.hasSubFolders && aFolder.hasFolderOrSubfolderNewMessages)
diff --git a/mailnews/mailnews.js b/mailnews/mailnews.js
--- a/mailnews/mailnews.js
+++ b/mailnews/mailnews.js
@@ -871,16 +871,18 @@ pref("mail.biff.on_new_window", true);
 // messages, as per the classic Thunderbird definition. Defaults to false, which
 // notifies about the number of unread messages.
 pref("mail.biff.use_new_count_in_badge", false);
 #endif
 #ifdef XP_WIN
 pref("mail.biff.use_new_count_in_badge", true);
 #endif
 
+pref("mail.biff.show_for_server", true);
+
 // For feed account serverType=rss sound on biff; if true, mail.biff.play_sound.* settings are used.
 pref("mail.feed.play_sound", false);
 
 // Content disposition for attachments (except binary files and vcards).
 //   0= Content-Disposition: inline
 //   1= Content-Disposition: attachment
 pref("mail.content_disposition_type", 1);
 
